### ISSUE TITLE
💅 `useImportType` error count does not match the number of errors outputted

### ISSUE THREAD

**Issue by aramikuto** [biomejs/biome#2048]
### Environment information

```bash
CLI:
  Version:                      1.6.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.19.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.14.3"

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### Rule name

useImportType

### Playground link

N/A

### Expected result

The number of errors found in the output of both the `check` and `lint` commands should match the number of errors that were printed.

Let's create a file named `a.ts` with the following content:
```tsx
import React from "react";
const a: React.ReactNode = null;
```

When executing the `biome lint a.ts` command, the output is:
```
a.ts:1:1 lint/style/useImportType  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ All these imports are only used as types.
  
  > 1 │ import React from "react";
      │ ^^^^^^^^^^^^^^^^^^^^^^^^^^
    2 │ const a: React.ReactNode = null;
    3 │ 
  
  ℹ Importing the types with import type ensures that they are removed by the transpilers and avoids loading unnecessary modules.
  
  ℹ Safe fix: Use import type.
  
    1 │ import·type·React·from·"react";
      │        +++++                   

Checked 1 file in 15ms. No fixes needed.
Found 2 errors.
lint ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.

```
It says `Found 2 errors`, but only one error is present.

---

When executing the `biome check a.ts` command, the output is:
```
a.ts:1:1 lint/style/useImportType  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ All these imports are only used as types.
  
  > 1 │ import React from "react";
      │ ^^^^^^^^^^^^^^^^^^^^^^^^^^
    2 │ const a: React.ReactNode = null;
    3 │ 
  
  ℹ Importing the types with import type ensures that they are removed by the transpilers and avoids loading unnecessary modules.
  
  ℹ Safe fix: Use import type.
  
    1 │ import·type·React·from·"react";
      │        +++++                   

Checked 1 file in 10ms. No fixes needed.
Found 3 errors.
check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.

```

The printed errors are the same as in the `lint` case, but the error count is 3 for some reason now.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Sec-ant** [comment]
Hello @chansuke, I did some debug for this issue just out of curiosity and for getting familiar with the codebase. The extra count of the error seems from this piece:

https://github.com/biomejs/biome/blob/984626de115e4b8f64dec97d3f66ba353ebf1b65/crates/biome_cli/src/execute/traverse.rs#L418-L420

The way I debugged it is printing a message before each line of `self.errors.fetch_add` and ran the `lint` and `check` commands respectively. Here are the results.

- `cargo run --bin biome lint ./a.ts --verbose --diagnostic-level=info`

  It says it found 2 errors, 1 of them is from the `Message::Failure` branch.

  ![image](https://github.com/biomejs/biome/assets/10386119/21512816-3d6c-43ee-8f2e-077611596be7)

- `cargo run --bin biome check ./a.ts --verbose --diagnostic-level=info`

  It says it found 3 errors, 2 of them are from the `Message::Failure` branch.

  ![image](https://github.com/biomejs/biome/assets/10386119/ca4e025f-cd6f-4706-8b80-498dd4e8f325)

I think the reason is that the error count will be correctly increased in other dedicated branches, so the one in branch `Message::Failure` seems redundant. I not sure though if we should remove this branch. But by searching `Message::Failure` in the whole codebase, it seems that this message will only be sent from the `lint` and `check` processes.

I hope my findings can provide some help :)

**Comment by chansuke** [comment]
@Sec-ant Hi!Thank you so much for the information!I was busy last week but start working on this issue tomorrow.

**Comment by Sec-ant** [comment]
Hi @chansuke, I'm gonna take over this issue to ship a fix before v1.6.4 is released.

**Comment by chansuke** [comment]
@Sec-ant 
NP. Thank you for your great work!!!
